test/mock-activity-port.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright 2023 The Subscribe with Google Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * This mock exists for unit tests.
+ */
+export class MockActivityPort {
```

Comment:
Should we make these new mock classes implement the interface they're mocking?

Reply:
Since the tests don't need that, I don't think it would really be worth it

Reply:
The current JavaScript code just returns `undefined` for all methods

Reply:
Can we back each of these methods with a publicly settable field? They can default to `undefined` but let users assign an override for convenience in testing.

Reply:
This is what I would expect from the test implementation, but we can defer that as a future enhancement instead of adding that to this migration.